src/main/java/org/opentripplanner/middleware/controllers/api/OtpUserController.java

```diff
@@ -88,23 +91,35 @@ protected OtpUser getUserProfile(Auth0UserProfile profile) {
      * HTTP endpoint to send an SMS text to an {@link OtpUser}'s phone number with a verification code. This is used
      * during user signup (or if a user wishes to change their notification preferences to use a new un-verified phone
      * number).
+     * Before sending a SMS request, this endpoint also saves the submitted phone number, marks it as not verified.
      */
     public VerificationResult sendVerificationText(Request req, Response res) {
         OtpUser otpUser = getEntityForId(req, res);
-        if (otpUser.phoneNumber == null) {
-            logMessageAndHalt(req, HttpStatus.NOT_FOUND_404, "User must have valid phone number to verify.");
+        // Get phone number from the path param.
+        String phoneNumber = req.params(PHONE_PARAM);
+        if (StringUtils.isBlank(phoneNumber)) {
+            logMessageAndHalt(req, HttpStatus.BAD_REQUEST_400, "A phone number must be provided.");
```

Comment:
After further inspection, this should probably validate the phone number against a regex (or some other validator) instead of just checking whether it's blank.

Reply:
I think Twilio throws an error if the number is not in E.164 (e.g. doesn't have the +1 prefix for US numbers).